pool/pool.go

```diff
@@ -505,17 +554,35 @@ func (c *clientStatusMonitor) address() string {
 	return c.addr
 }
 
-func (c *clientStatusMonitor) errorRate() uint32 {
-	return c.errorCount.Load()
+func (c *clientStatusMonitor) incErrorRate() {
+	c.currentErrorCount.Inc()
+	c.overallErrorCount.Inc()
+}
+
+func (c *clientStatusMonitor) currentErrorRate() uint32 {
+	return c.currentErrorCount.Load()
+}
+
+func (c *clientStatusMonitor) overallErrorRate() uint64 {
+	return c.overallErrorCount.Load()
 }
 
 func (c *clientStatusMonitor) resetErrorCounter() {
-	c.errorCount.Store(0)
+	c.currentErrorCount.Store(0)
+}
+
+func (c *clientStatusMonitor) latency() time.Duration {
+	return time.Duration(c.allTime.Load() / c.allRequests.Load())
```

Comment:
Can it ever panic because `allRequests` is 0?

Reply:
Also, what is the meaning of this? If we use the same counters for different operations, we have no useful information:
latency increase can mean network slowdown as well as increase in the number of `Put` operations.

Reply:
> Also, what is the meaning of this? If we use the same counters for different operations, we have no useful information: latency increase can mean network slowdown as well as increase in the number of `Put` operations.

It's a general latency in the first approximation. As I understand currently we have not  clear metrics

Reply:
What about using a single most frequent (GET ?) operation or a fixed-length array (for get/put/sth else?) operations?
This way we can provide an array of weights to have any metric we could imagine, while also having meaningful numbers.

So `0.9 * get_latency + 0.1 * put_latency` makes sense: this metric can be used by a client who has 90% GET requests and 10% PUT requests.